File: ProblemSolving/YetAnotherDeck.cpp
#include "YetAnotherDeck.h"
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;


string FindAnswers(vector<int> & cards, vector<int> & queries)
{
    string output = "";
    int q = queries.size();
    for (int i = 0; i < q; i++)
    {
        int pos = find(cards.begin(), cards.end(), queries[i]) - cards.begin();
        output += to_string(pos + 1) + " ";

        rotate(cards.begin(), cards.begin() + pos, cards.begin() + pos + 1);
    }

    return output;
}

YetAnotherDeck::YetAnotherDeck()
{
    int n, q;
    
    cin >> n >> q;

    vector<int> cards(n);

    for (size_t i = 0; i < n; i++)
    {
        cin >> cards[i];
    }

    vector<int> queries(q);

    for (size_t i = 0; i < q; i++)
    {
        cin >> queries[i];
    }
    
    cout << FindAnswers(cards, queries);
}

File: DotNetSolutions/YetAnotherDeck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DotNetSolutions
{
    //https://codeforces.com/contest/1511/problem/C
    class YetAnotherDeck
    {
        public YetAnotherDeck()
        {
            int n, q;
            string[] tokens = Console.ReadLine().Split();

            List<string> output = new List<string>();
            string[] strCards = Console.ReadLine().Split();
            int[] cards = Array.ConvertAll(strCards, int.Parse);

            string[] strQueries = Console.ReadLine().Split();
            int[] queries = Array.ConvertAll(strQueries, int.Parse);

            output.Add(FindAnswers(cards, queries));

            foreach (var val in output)
            {
                Console.Out.WriteLine(val);
            }
        }

        private void rotate (ref int[] cards, int pos)
        {
            int first = cards[pos];
            for (int i = pos; i > 0; --i)
            {
                cards[i] = cards[i - 1];
            }
            cards[0] = first;
        }


        private string FindAnswers(int[] cards, int[] queries)
        {
            string output = "";

            for (int i = 0; i < queries.Length; i++)
            {
                int pos = Array.IndexOf(cards, queries[i]);
                output += (pos + 1).ToString() + " ";

                rotate(ref cards, pos);
            }

            return output;
        }

    }
}
